File: tests/fixtures/typescript/simple-functions.ts
// Simple function with return type
function greet(name: string): string {
  console.log("Greeting: " + name);
  return `Hello, ${name}!`;
}

// Arrow function with expression body
const add = (a: number, b: number): number => a + b;

// Arrow function with block body
const multiply = (a: number, b: number): number => {
  const result = a * b;
  console.log("Result: " + result);
  return result;
};

// Async function
async function fetchData(url: string): Promise<string> {
  console.log("Fetching: " + url);
  const response = await fetch(url);
  return response.text();
}

// Function with default parameters
function createUser(name: string, age: number = 25, role: string = "user"): object {
  return { name, age, role };
}

// Exported function
export function formatDate(date: Date): string {
  return date.toISOString();
}

File: tests/fixtures/python/simple-functions.py
# Simple function with positional args
def greet(name):
    print("Hello, " + name)
    return "Hello, " + name


# Function with keyword arguments and defaults
def create_user(name, age=25, role="user"):
    return {"name": name, "age": age, "role": role}


# Function with *args and **kwargs
def log_message(*args, **kwargs):
    for arg in args:
        print(arg)
    for key, value in kwargs.items():
        print(f"{key}={value}")


# Function with type annotations
def add(a: int, b: int) -> int:
    return a + b


# Async function
async def fetch_data(url: str) -> str:
    print("Fetching: " + url)
    return ""
